src/dfcx_scrapi/builders/intents.py

```diff
@@ -680,3 +597,92 @@ def remove_label(self, label: Union[Dict[str, str], str]) -> Intent:
             )
 
         return self.proto_obj
+
+
+
+@dataclass
+class IntentStats():
+    """A class for tracking the stats of CX Intent object."""
+    intent_proto_obj: Intent
+
+    # Training Phrases
+    repeat_count_dict: Dict[int, int] = field(
+        default_factory=lambda: defaultdict(int)
+    )
+    annotated_count: int = 0
+    phrases_count: int = 0
+    annotated_pct: int = 0
+    uniques_count: int = 0
+    uniques_pct: int = 0
+    non_uniques_count: int = 0
+    non_uniques_pct: int = 0
+
+    # Parameters
+    params_count: int = 0
+
+
+    def calc_tps_stats(self):
+        """Calculating stats related to training phrases."""
+        for tp in self.intent_proto_obj.training_phrases:
+            self.repeat_count_dict[tp.repeat_count] += 1
+
+            for part in tp.parts:
+                if part.parameter_id:
+                    self.annotated_count += 1
+                    break
+
+
+        self.phrases_count = len(self.intent_proto_obj.training_phrases)
+        self.uniques_count = self.repeat_count_dict[1]
+        self.non_uniques_count = self.phrases_count - self.uniques_count
+
+        annot_pct = 100 * (self.annotated_count / self.phrases_count)
```

Comment:
@MRyderOC Edge case to consider.

If the Intent exists but has 0 Training Phrases, this method will fail and throw a `division by zero` error. 
Set up a simple check for count of `self.phrases_count` before diving into the calcs to prevent this.

"Why would anyone have an Intent with no Training Phrasees?" you might ask. 
It happens WAY MORE OFTEN than you'd think! 😆 

I think mostly when teams create placeholder Intents and don't add any TPs.